internal/flex/framework_test.go

```diff
@@ -964,3 +964,119 @@ func TestStringValueToFrameworkLegacy(t *testing.T) {
 		})
 	}
 }
+
+func TestFloat64ToFramework(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		input    *float64
+		expected types.Float64
+	}
+	tests := map[string]testCase{
+		"valid float64": {
+			input:    aws.Float64(42.1),
+			expected: types.Float64Value(42.1),
+		},
+		"zero float64": {
+			input:    aws.Float64(0),
+			expected: types.Float64Value(0),
+		},
+		"nil float64": {
+			input:    nil,
+			expected: types.Float64Null(),
+		},
+	}
+
+	for name, test := range tests {
+		name, test := name, test
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := Float64ToFramework(context.Background(), test.input)
+
+			if diff := cmp.Diff(got, test.expected); diff != "" {
+				t.Errorf("unexpected diff (+wanted, -got): %s", diff)
+			}
+		})
+	}
+}
+
+func TestFloat64ToFrameworkLegacy(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		input    *float64
+		expected types.Float64
+	}
+	tests := map[string]testCase{
+		"valid int64": {
+			input:    aws.Float64(42.1),
+			expected: types.Float64Value(42.1),
+		},
+		"zero int64": {
+			input:    aws.Float64(0),
+			expected: types.Float64Value(0),
+		},
+		"nil int64": {
+			input:    nil,
+			expected: types.Float64Value(0),
+		},
+	}
+
+	for name, test := range tests {
+		name, test := name, test
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := Float64ToFrameworkLegacy(context.Background(), test.input)
+
+			if diff := cmp.Diff(got, test.expected); diff != "" {
+				t.Errorf("unexpected diff (+wanted, -got): %s", diff)
+			}
+		})
+	}
+}
+
+func TestSet_Difference_strings(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		original Set[string]
+		new      Set[string]
+		expected Set[string]
+	}
+	tests := map[string]testCase{
+		"nil": {
+			original: nil,
+			new:      nil,
+			expected: nil,
+		},
+		"equal": {
+			original: Set[string]{"one"},
+			new:      Set[string]{"one"},
+			expected: nil,
+		},
+		"difference": {
+			original: Set[string]{"one", "two", "four"},
+			new:      Set[string]{"one", "two", "three"},
+			expected: Set[string]{"four"},
+		},
+		"difference_remove": {
+			original: Set[string]{"one", "two"},
+			new:      Set[string]{"one"},
+			expected: Set[string]{"two"},
+		},
```

Comment:
Maybe one more case for add only here? "difference" kind of covers it, but just to be explicit that nil is returned in that situation.